library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java

```diff
@@ -234,6 +241,8 @@ public GafferPopGraph(final Configuration configuration, final Graph graph) {
                 .dataAuths(configuration().getStringArray(DATA_AUTHS))
                 .build();
 
+        getAllElementsLimit = configuration.getInteger(GET_ALL_ELEMENTS_LIMIT, null);
```

Comment:
I think it would be useful to have a set default, instead of defaulting to `null`. With no default we'll need to make changes to various properties files and it will raise the barrier to entry when setting up - compared to using a sensible default. Gaffer also uses defaults everywhere else in these situations.

Reply:
Can you suggest a sensible default?

Reply:
Probably fairly high, say 1000. Although I'd also like others input. What do you think @tb06904?

Reply:
I'm slightly on the fence, my only worry with a default limit is awareness in that it would be very easy for a user to be unaware there is a default limit. The config would only be relevant to the gremlin side which requires a separate config anyways so no changes for existing graphs.

Reply:
As long as this is well documented, I think it should be fine.

Reply:
Fair point, I'd go a little higher as the default though maybe more like 5000? As a best of both worlds if possible could maybe look into throwing an exception if the limit is reached so the user is at least aware when it happens.